guava/src/com/google/common/math/IntMath.java

```diff
@@ -155,10 +162,14 @@ public static int log2(int x, RoundingMode mode) {
    *     is not a power of ten
    */
   @GwtIncompatible // need BigIntegerMath to adequately test
-  @SuppressWarnings("fallthrough")
-  public static int log10(int x, RoundingMode mode) {
+  @SuppressWarnings({"fallthrough",
+          "lowerbound:assignment.type.incompatible",/* line 172: only time when logFloor = log10Floor(x) is negative is
+          when `x` is 0, and that can't be because log10() only takes in positive `x` */
+          "upperbound:assignment.type.incompatible"/* since all elements in `maxLog10ForLeadingZeros` array
+          is @LTLengthOf("powersOf10","halfPowersOf10"), log10Floor() always return indexed values for `powersOf10` and `halfPowersOf10` */})
```

Comment:
If that is true, then why not annotate the return type of `log10Floor` to be `@IndexFor(value = {"powersOf10", "halfPowersOf10"})`? It is like that in `LongMath`.